sys fs/bees: update patchset for better performance and memory behavior

kakra opened the pull request:
This updates the patchset for better memory fragmentation behavior and also adds a patch to reduce logging pressure if a lot of new files have been added. The latter patch is pending upstream.

<!-- Please put the pull request description above -->

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

juippis reviewed (commented):
Cant' really make these changes to a stable ebuild. Please revbump 0.10 by adding a new copy, then reset keywords to unstable for that.

kakra commented:
Okay, I'll do that. I did it this way because `chkpkg` complained this should go into one single commit.

thesamesam commented:
What exactly did it say?

kakra commented:
> What exactly did it say?

First it said with a red message:
```
sys-fs/bees
  UnstableOnly: for arches: [ amd64, arm64 ], all versions are unstable: [ 0.10-r1 ]
  DroppedStableKeywords: commit e9d2511f191f (or later) dropped stable keyword: [ amd64 ]
```

So I added the stable keyword, and then it complained with a red message that there are extra commits and it should be a single commit instead. I reverted to the previous reflog commit instead now.

thesamesam commented on sys-fs/bees/bees-0.10.ebuild:
```diff
@@ -1,112 +0,0 @@
-	EGIT_REPO_URI="https://github.com/Zygo/bees.git"
-else
-	SRC_URI="https://github.com/Zygo/bees/archive/v${PV}.tar.gz -> ${P}.tar.gz"
-	KEYWORDS="amd64 ~arm64"
```
By adding 0.10-r1 as ~arch (which is correct) and then immediately dropping 0.10, you're removing all of bees' stable keywords.

See https://devmanual.gentoo.org/keywording/index.html#keywording-on-upgrades and the section below too.

kakra replied:
Ah okay, that's how to read that.

kakra replied:
I'll keep the drop commit for your convenience: It cleans up the old patch file. Either take it, or insert a stabilization commit before it.

thesamesam commented:
Yeah, the way to fix it was "add a new revision in ~arch _and don't cleanup the old one_", not do it all in one commit.

kakra commented:
Then use the cleanup commit in the way you need it. There's no point in keeping 0.10 once r1 is stable. It's just a micro patch that was edited to change how bees interacts with kernel memory allocation.

thesamesam commented:
Stabilisation should be done separately, ideally after 30 days where you file a bug for it, but it can be done sooner if urgent.

kakra commented:
Okay, so I'd create a bug for it to stabilize? And then I'd drop the last commit from here to apply it later?

thesamesam commented:
Exactly :)

kakra commented:
> Exactly :)

Thanks a lot for being explaining. Hope everything is fine now.

juippis commented on sys-fs/bees/bees-0.10-r1.ebuild:
```diff
@@ -0,0 +1,112 @@
+ERROR_BTRFS_FS="CONFIG_BTRFS_FS: bees does currently only work with btrfs"
+
+PATCHES=(
+	"${FILESDIR}/0001-context-demote-abandoned-toxic-match-to-debug-log-le.patch"
+	"${FILESDIR}/0002-HACK-crucible-Work-around-kernel-memory-fragmentatio_v2.patch"
```
We like to include `${PN}` in the patch files so they're more easily located.

kakra replied:
I'll do that with the next bump. What if multiple versions share the same patch?

juippis replied:
Then you just apply the same patch by hardcoding version.
`PATCHES=( "${FILESDIR}"/bees-0.10-fix-this-and-that.patch )`

juippis reviewed (approved):
Thanks!
